### PR TITLE
Update alloc-util.h

### PR DESCRIPTION
**Pull request by Dixiashui**
a process that have multi-thread senario, that in a thread ,the user uses 'sd_bus_default_system()' to create a new dbus connetion then uses  'sd_bus_default_flush_close()' to close the connection, the global variable ‘default_system_bus’ cann't be set to NULL, user will be not able to use  'sd_bus_default_system()' to create a dbus connection again in the process. the same when using 'sd_event_default()' due to this basic function 'mfree()'.
more, I can not find a reason that the space is freed but the pointer is not set to NULL.

### CODE REVIEW

**Comment by Dixiashui** [comment]
as pre description

**Comment by vcaputo** [comment]
Thanks for the contribution but your change basically doesn't do anything.

Could you instead create an issue describing the problem you're trying to solve and how to reproduce it?

**Comment by keszybz** [comment]
Yeah. This patch has no effect. Please open an issue.

**Comment by Dixiashui** [comment]
@vcaputo @keszybz 
-use sd_bus_default_system() create a dbus connection
-use sd_bus_default_flush_close() totally close the dbus connection
 but actually this function can't totally finish its work, the function didn't release the global variable( named 'default_system_bus' declared in the sd-bus.c file ) value.
-then use sd_bus_default_system() create a dbus connection in the process context,  it return create successfully but actually it is failed.
invoke stack:
sd_bus_default_system()
--bus_default()
and in the function bus_default() there is a code section:
       if (*default_bus) {
                *ret = sd_bus_ref(*default_bus);
                return 0;
        }
actually the *default_bus is not exist, but it's value is not clear by the 'sd_bus_default_flush_close()' deeply due to the 'mfree()'.

in summary 
 we can't loop create bus connection though
  -sd_bus_default_system() ->sd_bus_default_flush_close()->sd_bus_default_system()[**here will failed**]
in the same process context.

**Comment by keszybz** [comment]
Yep, that looks to be a bug. It looks like the global variable needs to be reset to NULL. WIll you be working on a patch?

**Comment by Dixiashui** [comment]
@keszybz
yes, and the series about event function (like sd_event_default())related to  the global variable "default_event" has the same issue due to the same reason.
I just want to fix it in this patch to creating this pull request, but it was refused and closed.

**Comment by keszybz** [comment]
The patch had no effect. A different patch that resets `default_system_bus` to `NULL` is needed.

**Comment by Dixiashui** [comment]
@keszybz 
checked my patch code,  yes, you are right.
